.claude/skills/twelve-factor-methodology/agents/agent.code-scan.sh

```
@@ -0,0 +1,84 @@
+# Anti-pattern detection
+# ============================================================================
+VIOLATIONS=()
+
+# Check for hardcoded credentials
+if grep -rn "password\s*=\s*['\"][^$]" --include="*.py" --include="*.js" --include="*.rb" "$ROOT" 2>/dev/null | head -5; then
+  VIOLATIONS+=($(jq -cn '{
+    "type": "hardcoded-credentials",
+    "severity": "critical",
+    "factor": "III. Config",
+    "description": "Hardcoded credentials detected"
+  }'))
```

Shellcheck (0.11.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)